Fix Inserter: Navigate between blocks with left/right

Soean opened the pull request:
## Description
To navigate in NavigableContainer you can now use orientation="both" and use the left/up keys for the previous item and right/down keys for the next item.

Fixes #3850

Soean committed:
Same key behavior for horizontal and vertical.

Soean committed:
Adjust the tests

Soean committed:
Remove orientation from tests

youknowriad commented on components/navigable-container/index.js:
```diff
@@ -124,18 +124,10 @@ export class NavigableMenu extends Component {
 		const eventToOffset = ( evt ) => {
 			const { keyCode } = evt;
 
-			const isVertical = orientation === 'vertical';
```
I guess this means the "orientation" prop is useless now and we should remove all its references (docs, code...).
I'm not against this personally, we just have to make sure it's something we want for all `NavigableContainer` usage

youknowriad replied:
@jasmussen Are we ok with this change? It makes `left` and `up` / `down` and `right` work similarily in the inserter and the toolbar and the dropdowns?

Soean replied:
I removed it in d4df52a, but not sure if it is needed anywhere.

jasmussen replied:
Looking at the behavior in `master`, I do agree that it's a little weird that pressing _down_ goes _right_ (depending on where you are), whereas pressing _right_ does nothing. As such this _seems_ like a good change. But it also begs the question why down doesn't go down, and right doesn't go right. 

But it is the inserter, and there's a rich history in how we got here, so before I say anything definitive I'd love @afercia's thoughts.

afercia replied:
We've received some feedback during WCUS contributor day while testing the inserter, same confusion about _which_ arrows where expected to work. I think the arrow navigation should allow three "modes", depending on the nature of the UI component:
- horizontal: LEFT and RIGHT to go previous and next horizontally, e.g. toolbars
- vertical: UP and DOWN to go previous and next vertically, e.g. single column menus
- both: "spatial" navigation where UP and LEFT go previous, and DOWN and RIGHT go next

And yes, in the Inserter I'd use "both" 🙂

youknowriad replied:
@Soean Maybe instead of removing the "orientation" entirely, we keep it and add a third option "both" or "vertical-horizontal". That way we could enable this behavior in the inserter but still keep it as is for the BlockToolbar and the Dropdowns. The Dropdown should probably be only "vertical" while the block toolbar should be "horizontal". Thoughts?

Soean replied:
@youknowriad I had a similar thought. In WAI-ARIA 1.1 is a new default value `undefined: The element's orientation is unknown/ambiguous.` https://www.w3.org/TR/wai-aria-1.1/#aria-orientation

Soean committed:
Remove orientation

Soean committed:
Revert "Remove orientation"

This reverts commit d4df52a13dadf63fa26bd02fee12ae04365f470e.

Soean committed:
Revert "Remove orientation from tests"

This reverts commit a1f9fcf01aca0075452115c3f48cfcb4e83466cc.

Soean committed:
Revert "Adjust the tests"

This reverts commit 0a1e13ca720b47eab4b783c1753d9c110af7d824.

Soean committed:
Revert "Same key behavior for horizontal and vertical."

This reverts commit 3240f96d7e4844bdec31b39e8adc1112d459484c.

Soean committed:
Add orientation undefined for the inserter.

youknowriad commented on editor/components/inserter/group.js:
```diff
@@ -77,7 +77,7 @@ export default class InserterGroup extends Component {
 			<NavigableMenu
 				className="editor-inserter__category-blocks"
-				orientation="vertical"
+				orientation="undefined"
```
Minor: I know this is the equivalent of the aria value, but it's weird. I'd prefer something like "all" personally

youknowriad reviewed (approved):
Can we have a unit test for this orientation?

Anyway, LGTM 👍

Soean committed:
Change 'undefined' to 'all'

Soean committed:
Change 'all' to 'both' and add docs

Soean committed:
Pass old tests

Soean committed:
Add Unit test for 'both'

Soean commented:
@youknowriad Thanks for your feedback. 
- I changed `undefined` to `both`, because there are just two directions.
- I added a test for `both`.
- I added `both` to the docs.

youknowriad commented:
Nice, might be better to wait for the release before merging
